pyiron_gui/project/project_browser.py

```diff
@@ -140,35 +272,68 @@ def __getattr__(self, item):
     def clear_output(self, *args, **kwargs):
         clear_button = kwargs.pop('clear_button', False)
         if clear_button:
-            self.buttons = widgets.HBox()
+            self.header = widgets.HBox()
         self.output.clear_output(*args, **kwargs)
         self.refresh()
 
-    def _update_buttons(self, obj):
+    def _click_button(self, b):
+        self.output.clear_output()
+        if b.description == "Show data":
+            b.description = "Show plot"
+            b.tooltip = "Display the plotting representation"
+            self.header.children = tuple([b])
+            self.refresh()
+            with self.output:
+                display(b.obj)
+        elif b.description == "Re-plot":
+            with self.output:
+                plt.ioff()
+                display(self._plot_array(b.obj))
+        else:
+            self.display(obj=b.obj)
 
-        def click_button(b):
-            self.output.clear_output()
-            self.display(obj=obj)
+    def _update_header(self, obj):
+        childs = []
 
-        if isinstance(obj, PyironWrapper) and obj.has_self_representation:
+        button = None
+        if isinstance(obj, BaseWrapper) and hasattr(obj, 'option_representation'):
+            childs.append(obj.option_representation)
+            button = widgets.Button(description="Apply")
```

Comment:
Looping back to my earlier comment about strings and button power, I see now how the strings are being used to set the button behaviour and remain uncomfortable. IMO it will be more stable to have something like this:

```python
button = my_widgets.ApplyButton()
```

And later

```python
button = my_widgets.ReplotButton(name=obj.name)
```